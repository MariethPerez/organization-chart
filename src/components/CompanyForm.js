/** @jsx jsx */
import React from "react";
import { jsx } from "@emotion/core";

function CompanyForm({ setCompanyName }) {
  const formContainer = {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    height: "100vh"
  };

  const styledForm = {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    width: "500px",
    padding: "1.5em",
    background: "black",
    borderRadius: "0.5em"
  };

  const styledHeader = {
    textAlign: "center",
    margin: "15px",
    fontSize: "40px",
    color: "white",
    fontWeight: "bolder"
  };

  const inputCompanyName = {
    margin: "1em 0",
    fontSize: "30px",
    paddingTop: "10px",
    paddingBottom: "10px",
    borderRadius: "4px",
    outline: "none",
    border: "none",
    borderBottom: "1px solid #ddd",
    width: "100%",
    textAlign: "center",
    '&:focus': {
      border: "1px solid red"
    }
  };

  const submitButton = {
    lineHeight: "1.5em",
    background: "darkgray",
    border: "none",
    color: "white",
    fontSize: "20px",
    fontWeight: "bold",
    padding: "0.5em",
    cursor: "pointer",
    borderRadius: "0.25rem",
    textAlign: "center",
    width: "50%"
  };

  const [content, setCont] = React.useState("");

  function handleSubmit(event) {
    event.preventDefault();
    setCompanyName(content);
  }

  function handleChange(event) {
    setCont(event.target.value);
  }

  return (
    <div css={formContainer}>
      <form onSubmit={handleSubmit} css={styledForm}>
        <header css={styledHeader}>Company's name</header>
        <input
          css={inputCompanyName}
          aria-label="Input to get Company's name"
          placeholder="Enter company's name"
          type="text"
          required
          name="company"
          id="company"
          onChange={handleChange}
          autoFocus
          autoComplete="off"
        />
        <button css={submitButton} type="submit">Next</button>
      </form>
    </div>
  );
}

export default CompanyForm;
