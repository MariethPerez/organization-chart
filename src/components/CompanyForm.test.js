import React from "react";
import { render, fireEvent } from "@testing-library/react";

import CompanyForm from "./CompanyForm";

test("CompanyForm functionality", () => {
  const setCompanyName = jest.fn();
  const { asFragment, getByLabelText } = render(
    <CompanyForm setCompanyName={setCompanyName} />
  );

  expect(asFragment()).toMatchSnapshot();
  const input = getByLabelText("Input to get Company's name");

  fireEvent.change(input, {
    target: { value: "Able" }
  });

  fireEvent.submit(input);

  expect(setCompanyName).toHaveBeenCalledWith("Able");
});
