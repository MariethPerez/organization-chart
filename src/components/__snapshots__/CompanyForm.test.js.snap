// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyForm functionality 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
  padding: 1.5em;
  background: black;
  border-radius: 0.5em;
}

.emotion-0 {
  text-align: center;
  margin: 15px;
  font-size: 40px;
  color: white;
  font-weight: bolder;
}

.emotion-1 {
  margin: 1em 0;
  font-size: 30px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 4px;
  outline: none;
  border: none;
  border-bottom: 1px solid #ddd;
  width: 100%;
  text-align: center;
}

.emotion-1:focus {
  border: 1px solid red;
}

.emotion-2 {
  line-height: 1.5em;
  background: darkgray;
  border: none;
  color: white;
  font-size: 20px;
  font-weight: bold;
  padding: 0.5em;
  cursor: pointer;
  border-radius: 0.25rem;
  text-align: center;
  width: 50%;
}

<div
    class="emotion-4"
  >
    <form
      class="emotion-3"
    >
      <header
        class="emotion-0"
      >
        Company's name
      </header>
      <input
        aria-label="Input to get Company's name"
        autocomplete="off"
        class="emotion-1"
        id="company"
        name="company"
        placeholder="Enter company's name"
        required=""
        type="text"
      />
      <button
        class="emotion-2"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</DocumentFragment>
`;
